mlrun/feature_store/api.py

```diff
@@ -492,7 +493,21 @@ def ingest(
         featureset.spec.source = source
         featureset.spec.validate_no_processing_for_passthrough()
 
-    namespace = namespace or get_caller_globals()
+    if not namespace:
+        if run_config and run_config.local and run_config.function:
+            filename = run_config.function.spec.filename
+            if filename:
+                import importlib.util
+                import sys
+
+                module_name = pathlib.Path(filename).name.rsplit(".", maxsplit=1)[0]
+                spec = importlib.util.spec_from_file_location(module_name, filename)
+                module = importlib.util.module_from_spec(spec)
+                sys.modules[module_name] = module
+                spec.loader.exec_module(module)
+                namespace = vars(__import__(module_name))
+        else:
+            namespace = get_caller_globals()
```

Comment:
worth adding a description and comment with what is intended to be done here. 
another suggestion is to move that logic to a method, this ingest method is already too long and shouldn't hold all the code logic as well, this will help us further in the future with the all client server separation.

Reply:
Done